### ISSUE TITLE
[editor] annoying auto-suggestion when writing comments

### ISSUE THREAD

**Issue by vince-fugnitto** [eclipse-theia/theia#5227]
**Description**

There is an annoying auto-suggestion feature when writing comments.
I do not expect suggestions when writing comments as it is intrusive to the experience.

<div align='center'>

![bug](https://user-images.githubusercontent.com/40359487/58249466-f9884080-7d2c-11e9-854d-5380789611fc.gif)

</div>

In comparison, here's vscode:

<div align='center'>

![vscode](https://user-images.githubusercontent.com/40359487/58249560-3a805500-7d2d-11e9-8ac3-d56c08e4934c.gif)

</div>

**Comment by sauny** [comment]
Have been looking into this and from what I can see the problem is within packages/monaco/src/browser/monaco-snippet-suggest-provider.ts.  I can intercept when the snippets are provided but I am having difficulty working out if it is within a comment block.  The only feasible way I can see to do this is to the class name of the div what is being written... any other ideas?

**Comment by vince-fugnitto** [comment]
@sauny I took a look at the hint you provided and saw that `provideCompletionItems` is the one that provides the snippets from the bug.

https://github.com/theia-ide/theia/blob/c8dc1398fe1c9427f2d28b3c16049eeab8f6cb80/packages/monaco/src/browser/monaco-snippet-suggest-provider.ts#L30

For test purposes updating the code to return `[]` fixed the bug. I'm not entirely sure what the method is used for, since without it I am still getting proper snippets.

**Comment by vince-fugnitto** [comment]
@sauny I wonder if we can detect when we have a `comment` and not provide the snippet help.

**Comment by sauny** [comment]
@vince-fugnitto I thought that, the comments get registered in the grammar-contribution files but I see nowhere where they get used.  If I can work out where theia decides to set the CSS for comments, I should be able to set something to say it is in comment mode... but I can't find a way to do it....

**Comment by vince-fugnitto** [comment]
> @vince-fugnitto I thought that, the comments get registered in the grammar-contribution files but I see nowhere where they get used. If I can work out where theia decides to set the CSS for comments, I should be able to set something to say it is in comment mode... but I can't find a way to do it....

I don't quite understand what you mean about the CSS suggestion.

**Comment by sauny** [comment]
I meant the Monaco editor sets the css for a comment (it goes green). But I can't work out where that happens.

**Comment by BroKun** [comment]
I noticed this problem, it is caused by the configuration of monaco. 
```ts
// type in monaco editor
quickSuggestions?: boolean | {
	other: boolean;
	comments: boolean;
	strings: boolean;
};
// editor preference
'editor.quickSuggestions': {
       'type': 'boolean',
       'default': true,
       'description': 'Enable quick suggestions (shadow suggestions)'
},
```
The value `True` means you will see suggestions anywhere. 
We should not display suggestions by default in string and comment, just like vscode.

**Comment by akosyakov** [comment]
@BroKun can you handle it please? Please could you also reference a corresponding default configuration in VS Code, just to check that it is so.

**Comment by BroKun** [comment]
> @BroKun can you handle it please? Please could you also reference a corresponding default configuration in VS Code, just to check that it is so.


Yes, I am doing this. 
And I found that there is another problem, we need to pass the object that can be rewritten to `monaco`, especially when updating the configuration, but now the `preferrenceService` provides read-only objects. 
I will solve this problem first, but it is better if the `preferenceService` can support it. Maybe I should open another issue.

**Comment by BroKun** [comment]
@vince-fugnitto This seems to have been resolved after #5931 and Monaco upgraded to 0.17.0.

**Comment by vince-fugnitto** [comment]
> @vince-fugnitto This seems to have been resolved after #5931 and Monaco upgraded to 0.17.0.

I forgot to close, thank you!
